File: Bismo.java

import java.util.List;
import java.util.Scanner;

public class Bismo {
	private static String perintah = "";
	private static List<Bismo.Loker> lokers = new java.util.LinkedList<Bismo.Loker>();

	public static void main(String[] args) {
		Scanner i = new Scanner(System.in);
		System.out.println("Loker's Spelling");
		System.out.println("----------------");
		while (!perintah.equals("EXIT")) {
			System.out.print("--> ");
			perintah = i.nextLine();
			eksekusi();
		} i.close();
	}

	private static void eksekusi() {
		perintah = perintah.toUpperCase();
		if (perintah.equals("EXIT")) {
			System.out.println("Good bye! :-)");
		} else if (0 == lokers.size() && perintah.startsWith("INIT ") && 2 == bagiKata()) {
			initLokers();
		} else if (0 < lokers.size() && perintah.equals("STATUS")) {
			statusLokers();
		} else if (0 < lokers.size() && perintah.startsWith("INPUT ") && 3 == bagiKata()) {
			inputLokers();
		} else if (0 < lokers.size() && perintah.startsWith("LEAVE ") && 2 == bagiKata()) {
			leaveLokers();
		} else if (0 < lokers.size() && perintah.startsWith("FIND ") && 2 == bagiKata()) {
			findLokers();
		} else if (0 < lokers.size() && perintah.startsWith("SEARCH ") && 2 == bagiKata()) {
			srcLokers();
		} else System.out.println("Spell cannot be realize :-(");
	}

	private static void srcLokers() {
		String[] pecah = perintah.split(" ");
		boolean lanjut = false, kosong = true;
		for (Bismo.Loker l:lokers) if(pecah[1].equals(l.tipe)) {
			if (lanjut) System.out.print(", ");
			System.out.print(l.no);
			lanjut = true;
			kosong = false;
		} if (lanjut) System.out.println();
		if (kosong) System.out.println("Tipe kartu ini tidak tersimpan");
	}

	private static void findLokers() {
		String[] pecah = perintah.split(" ");
		try {
			long g = Long.parseLong(pecah[1]);
			boolean ketemu = false;
			for (Bismo.Loker l:lokers) if (g == l.no) {
				ketemu = true;
				System.out.println("Kartu identitas tersebut berada di loker nomor " + l.indek);
			} if (!ketemu) System.out.println("Kartu identitas tidak ditemukan");
		} catch (NumberFormatException e) {
			System.out.println("Sorry, spell params cannot be allowed :-(");
		}
	}

	private static void leaveLokers() {
		String[] pecah = perintah.split(" ");
		try {
			int g = Integer.parseInt(pecah[1]);
			Bismo.Loker l = lokers.get(g - 1);
			if (-1 < l.no && g == l.indek) {
				l.no = -1;
				l.tipe = "___";
				System.out.println("Loker nomor " + g + " berhasil dikosongkan");
			} else System.out.println("Loker nomor " + g + " tidak berisi");
		} catch (NumberFormatException e) {
			System.out.println("Sorry, spell params cannot be allowed :-(");
		}
	}

	private static void inputLokers() {
		String[] pecah = perintah.split(" ");
		try {
			long g = Long.parseLong(pecah[2]);
			boolean terisi = false;
			for (int i = 0; i < lokers.size(); i++) {
				Bismo.Loker l = lokers.get(i);
				if (l.no == -1) {
					l.no = g;
					l.tipe = pecah[1];
					terisi = true;
					System.out.println("Kartu identitas tersimpan di loker nomor " + l.indek);
					break;
				}
			} if (!terisi) System.out.println("Maaf loker sudah penuh");
		} catch (NumberFormatException e) {
			System.out.println("Sorry, spell params cannot be allowed :-(");
		}
	}

	private static void statusLokers() {
		System.out.println("No Loker\tTipe Identitas\tNo Identitas");
		for (Bismo.Loker l:lokers) System.out.println(l);
	}

	private static void initLokers() {
		String[] pecah = perintah.split(" ");
		try {
			int g = Integer.parseInt(pecah[1]);
			if (g > 0) {
				for (int i = 0; i < g; i++) lokers.add(new Bismo.Loker(i + 1, "___", -1));
				System.out.println("Berhasil membuat loker dengan jumlah " + g);
			} else System.out.println("Sorry, spell params cannot be allowed :-(");
		} catch(NumberFormatException e) {
			System.out.println("Sorry, spell params cannot be allowed :-(");
		}
	}

	private static int bagiKata() {
		String[] pecah = perintah.split(" ");
		return pecah.length;
	}

	public static class Loker {
		private int indek;
		private String tipe;
		private long no;

		public Loker() {
			super();
		}

		public Loker(int indek, String tipe, long no) {
			super();
			this.indek = indek;
			this.tipe = tipe;
			this.no = no;
		}

		public int getIndek() {
			return indek;
		}

		public void setIndek(int indek) {
			this.indek = indek;
		}

		public String getTipe() {
			return tipe;
		}

		public void setTipe(String tipe) {
			this.tipe = tipe;
		}

		public long getNo() {
			return no;
		}

		public void setNo(long no) {
			this.no = no;
		}

		@Override
		public String toString() {
			return "" + indek + "\t\t" + tipe + "\t\t" + no;
		}
	}
}

File: Bismo.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

type Loker struct {
	indek int
	no    int64
	tipe  string
}

var perintah string = ""
var lokers []Loker

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	fmt.Println("Loker's Spelling")
	fmt.Println("----------------")
	for strings.Compare(perintah, "EXIT") != 0 {
		fmt.Print("--> ")
		if scanner.Scan() {
			perintah = scanner.Text()
			eksekusi()
		}
	}
}

func eksekusi() {
	perintah = strings.ToUpper(perintah)
	if strings.Compare(perintah, "EXIT") == 0 {
		fmt.Println("Good bye :-)")
	} else if lokers == nil && strings.HasPrefix(perintah, "INIT ") && 2 == bagiKata() {
		initLokers()
	} else if lokers != nil && 0 == strings.Compare(perintah, "STATUS") {
		statusLokers()
	} else if lokers != nil && strings.HasPrefix(perintah, "INPUT ") && 3 == bagiKata() {
		inputLokers()
	} else if lokers != nil && strings.HasPrefix(perintah, "LEAVE ") && 2 == bagiKata() {
		leaveLokers()
	} else if lokers != nil && strings.HasPrefix(perintah, "FIND ") && 2 == bagiKata() {
		findLokers()
	} else if lokers != nil && strings.HasPrefix(perintah, "SEARCH ") && 2 == bagiKata() {
		srcLokers()
	} else {
		fmt.Println("Sorry, this spell not found :-(")
	}
}

func srcLokers() {
	var pecah = strings.Split(perintah, " ")
	var lanjut bool = false
	for i := 0; i < len(lokers); i++ {
		var l = lokers[i]
		if strings.Compare(pecah[1], l.tipe) == 0 {
			if lanjut {
				fmt.Print(", ")
			}
			fmt.Printf("%d", l.no)
			lanjut = true
		}
	}
	if lanjut {
		fmt.Println("")
	} else {
		fmt.Println("Tipe kartu ini tidak tersimpan")
	}
}

func findLokers() {
	var pecah = strings.Split(perintah, " ")
	var g, err = strconv.ParseInt(pecah[1], 0, 64)
	if err == nil && g > 0 {
		var ketemu bool = false
		for i, l := range lokers {
			if g == l.no {
				ketemu = true
				fmt.Printf("Kartu identitas tersebut berada di loker nomor %d\n", i+1)
			}
		}
		if !ketemu {
			fmt.Println("Kartu identitas tidak ditemukan")
		}
	} else {
		fmt.Println("Sorry, spell params cannot be allowed :-(")
	}
}

func leaveLokers() {
	var pecah = strings.Split(perintah, " ")
	var g, err = strconv.Atoi(pecah[1])
	if err == nil && g > 0 {
		if -1 != lokers[g-1].no && lokers[g-1].indek == g {
			lokers[g-1].no = -1
			lokers[g-1].tipe = "___"
			fmt.Printf("Loker nomor %d berhasil dikosongkan\n", g)
		} else {
			fmt.Printf("Loker nomor %d tidak berisi\n", g)
		}
	} else {
		fmt.Println("Sorry, spell params cannot be allowed :-(")
	}
}

func inputLokers() {
	var pecah = strings.Split(perintah, " ")
	var g, err = strconv.ParseInt(pecah[2], 0, 64)
	if err == nil && g > 0 {
		var terisi bool = false
		for i := 0; i < len(lokers); i++ {
			if lokers[i].no == -1 {
				lokers[i].no = g
				terisi = true
				lokers[i].tipe = pecah[1]
				fmt.Printf("Kartu identitas tersimpan di loker nomor %d\n", lokers[i].indek)
				break
			}
		}
		if !terisi {
			fmt.Println("Maaf loker sudah penuh")
		}
	} else {
		fmt.Println("Sorry, spell params cannot be allowed :-(")
	}
}

func statusLokers() {
	fmt.Println("No Loker\tTipe Identitas\tNo Identitas")
	for i := 0; i < len(lokers); i++ {
		l := lokers[i]
		fmt.Printf("%d\t\t%s\t\t%d\n", l.indek, l.tipe, l.no)
	}
}

func initLokers() {
	var pecah = strings.Split(perintah, " ")
	var g, err = strconv.Atoi(pecah[1])
	if err == nil && g > 0 {
		lokers = make([]Loker, g)
		for i := 0; i < g; i++ {
			lokers[i].indek = i + 1
			lokers[i].no = -1
			lokers[i].tipe = "___"
		}
		fmt.Printf("Berhasil membuat loker dengan jumlah %d\n", g)
	} else {
		fmt.Println("Sorry, spell params cannot be allowed :-(")
	}
}

func bagiKata() int {
	var pecah = strings.Split(perintah, " ")
	return len(pecah)
}
